include/linux/rwsem_compat.h

```diff
@@ -35,12 +35,19 @@
 #define	SPL_RWSEM_SINGLE_WRITER_VALUE	(RWSEM_ACTIVE_WRITE_BIAS)
 #endif
 
-/* Linux 3.16 change activity to count for rwsem-spinlock */
+/*
+ * Linux 3.16 changed activity to count for rwsem-spinlock
+ * Linux 4.8 changed count to an atomic_long_t for !rwsem-spinlock
+ */
 #ifdef HAVE_RWSEM_ACTIVITY
 #define	RWSEM_COUNT(sem)	sem->activity
 #else
+#ifdef HAVE_RWSEM_ATOMIC_LONG_COUNT
```

Comment:
Yeah, I went back and forth about doing this as an `#elif defined()`.  I'll refresh the patch with that change since after looking at the whole patch I think I like that better too.